.gitignore

```diff
@@ -0,0 +1,8 @@
+cmake.bld
+compile_commands.json
+/.build/
+/build/
+/.cache/
+*.pyg
+.update-submodules
+.vscode/
```

Comment:
```suggestion
```
Editor specific files belong in a user's global .gitignore file. I think this should list only those files that this repository will create as part of its build process. That'd be only the build directory, right?

Reply:
agree. please check again